contracts/contracts/othellofactory.sol

```diff
@@ -174,17 +177,22 @@ contract othellofactory is Board{
             }
         }
         removeUsersFromGame();
-        emit EndGame(blackaddress,whiteaddress);
         
-        if(white==black){
-            return(blackaddress, true);
-        }
-        winner=black>white?blackaddress:whiteaddress;
+        _isDraw=white==black?true:false;
+        _winner=black>=white?blackaddress:whiteaddress;
+        address _looser=_winner==blackaddress?whiteaddress:blackaddress;
+        emit EndGame(_winner,_looser);
 
-        return(winner, false);
+        return(_winner, _isDraw);
             
     }
     
+    // Returns the address of the player whose turn it is
+    function getCurrentTurnAddress() private view returns(address currentPlayer){
+        address currentTurnAddress=getMyGame().isWhiteTurn? getMyGame().whiteaddress: getMyGame().blackaddress;
+        return currentTurnAddress;
+    }
+    
     // Return the currentState of player's game
     function getCurrentState() public view returns (string memory state){
         if (activelyPlaying[msg.sender]==true){
```

Comment:
This i think causes an error as the function is declared as view.. please remove view